README.rst

```diff
@@ -1,102 +1,208 @@
+  well-typed. We thus aim for the expressivity of Template Haskell, not the
+  extra guarantees of Typed Template Haskell.
+* Purely functional; primitives with side-effects either run in the ``IO``
+  monad or in the ``Macro`` monad, as appropriate.
+* Higher-kinded types; for example `monads`_ are defined as a library.
+* `Algebraic datatypes`_. In Racket, ``(list 1 2 3)`` and ``'(1 2 3)``
+  construct the same value, an s-expression which satisfies the ``list?``
+  predicate. In Klister, the former produces a value of the algebraic datatype
+  ``(List Integer)``, while the latter produces a value of type ``Syntax``.
+  This syntax can be "opened", via ``(open-syntax '(1 2 3))``, to obtain the
+  value ``(list-contents (list '1 '2 '3))`` of type
+  ``(Syntax-Contents Syntax)``.
+* `Bidirectional type inference`_; the type of an argument can specialize the
+  type of the function being called.
+* Bidirectional type inference; the type of a function can specialize the type
+  of its arguments.
+
+.. _monads: TODO: link to monad.kl's Monad definition, and add a comment there
+   highlighting the inferred type, especially the higher-kinded type variable.
+.. _Algebraic datatypes: TODO: write a small example defining and matching on
+   an algebraic type. Perhaps Either?
+.. _Bidirectional type inference: TODO: write a small example demonstrating
+   those two features.
+
+Features which make Klister special (but not *unique*; see the `bibliography`_):
+
+* `Type-providing macros`_; a macro used in an argument position can provide
+  type information about the argument, and can thus specialize the type of the
+  function being called.
+* `Type-aware macros`_; a macro used in an argument position can obtain the
+  type of the argument it needs to generate, and thus the type of a function
+  can affect what arguments are generated.
+* `Stuck macros`_; it is thus possible for multiple macros to affect what each
+  other generates. The language primitives are designed so that the order in
+  which the macros are expanded cannot affect their results, and indeed the
+  same is true for the order in which the macro expansion and type-inference
+  steps are interleaved. This makes Klister code more robust to refactoring.
+* `Problem-aware macros`_; in addition to the type, a macro can learn which
+  "problem" it needs to solve, namely whether it must generate an expression, a
+  type, a pattern, etc.
+* `Custom type-driven code generation`_, via macros which generate code from a
+  type.
+* Languages with `custom type systems`_, via macros which reinterpret types
+  into those of an existing ``#lang``, and which contribute to type inference
+  by providing type information about the code they generate. The variety of
+  type-systems which can be implemented this way is unforunately limited by
+  the core type system to which everything must be rewritten.
+* Languages with `custom implicit terms`_, via macros which generate terms of
```

Comment:
I think this is not a feature, but rather a use of the features, right?

Reply:
fixed in bf9c31e3502a3e5f6b79ba79038422af6c6d6db9